modules/platform/nuxeo-platform-document-routing/nuxeo-routing-core/src/main/java/org/nuxeo/ecm/platform/routing/core/io/TaskWriter.java

```diff
@@ -108,7 +113,12 @@ public void writeEntityBody(Task item, JsonGenerator jg) throws IOException {
             if (StringUtils.isNotBlank(workflowInstanceId)) {
                 NodeAccessRunner nodeAccessRunner = new NodeAccessRunner(wrapper.getSession(), workflowInstanceId,
                         nodeId);
-                nodeAccessRunner.runUnrestricted();
+                try {
+                    nodeAccessRunner.runUnrestricted();
+                } catch (DocumentNotFoundException e) {
+                    // If the DocumentRoute is not found (deleted for example), the TaskWriter fails.
+                    log.warn(e, e);
```

Comment:
As @efge said in one of his comments, you should log something meaningful if possible:
```
log.warn("Failed to get workflow instance", e)
```
BTW, do we want the whole stack trace for a WARN? (cc @kevinleturc)

Reply:
I would say no as this is a valid error case, I'm wondering if we still want it behind a debug/trace level.

Reply:
So a simple:
```
log.warn("Failed to get workflow instance");
log.debug(e, e);
```
?

Reply:
Looks good 👍